File: bubble.c
#include<stdio.h>
#include<stdlib.h>
#define LEN 5

void bubble(int *a){
    for(int i=0;i<LEN;i++){
        for(int j=0;j<LEN-i-1;j++){
            if(*(a+j)>*(a+j+1)){
                int temp = *(a+j);
                *(a+j) = *(a+j+1);
                *(a+j+1) = temp;
            }
        }
    }
}

void print(int *a){
    for(int i=0;i<LEN;i++){
        printf("%i ",*(a+i));
    }
    puts("");
}

void main(){
    int *arr = (int *)malloc(LEN*sizeof(int));
    *arr = 22;
    *(arr+2) = 54;
    *(arr+4) = -72;
    printf("Unsorted Array: ");
    print(arr);
    bubble(arr);
    printf("Sorted Array: ");
    print(arr);
}

File: bubble.rb
def bubble a
    for i in 0...a.length
        for j in 0...a.length-i-1
            a[j],a[j+1] = a[j+1],a[j] if a[j] > a[j+1]
        end
    end
    a
end

p bubble [2,1,-3,4,17,-5,37,8]
